docs/docs/usage/scaladoc/site-versioning.md

```diff
@@ -0,0 +1,33 @@
+---
+title: Site versioning
+---
+
+# {{ page.title }}
+
+Scaladoc provides a convenient way to switch between different versions of the documentation. The feature is useful if we want to expose older docs for users that didn't migrate to the new version of our library.
+
+### How to setup it
+
+The feature was designed for easy scalability with no need to regenerate all scaladocs after adding a new version. To do so a new setting is introduced:  `-versions-dictionary-url`. Its argument must be a URL to a JSON document holding information about the locations of specific versions. The JSON file has single property `versions` that holds the dictionary associating the labels of specific versions of the documentation to the URLs pointing to their index.html
+
+Example JSON file:
+```
+{
+  "versions": {
+    "3.0.x": "https://dotty.epfl.ch/3.0.x/docs/index.html",
+    "Nightly": "https://dotty.epfl.ch/docs/index.html"
+  }
+}
+```
+
+This enforce us to provide the setting while generating docs for each of the versions, however it gives us more flexibility later. If you want to add a version of the API docs next to the previous 5 versions that you have already published, then you only need to upload the new docs to a web server and add a new entry to the JSON file. All versions of the site will now become aware of the new site version.
+
+The important thing to note is that there is only one JSON file to avoid redundancy and each scaladoc must set up it's URL location beforehand, for example:
+`doc / scalacOptions ++= Seq("-versions-dictionary-url", "https://dotty.epfl.ch/versions.json")`
```

Comment:
```suggestion
The important thing to note is that there is only one JSON file to avoid redundancy and each scaladoc must set up it's URL location beforehand, for example, in sbt:

~~~
doc / scalacOptions ++= Seq("-versions-dictionary-url", "https://dotty.epfl.ch/versions.json")
~~~
```

Reply:
What do you mean by “each scaladoc must set up its URL location _beforehand_”?

Reply:
Updated

Reply:
Well, I mean that for every version specific scaladoc, we have to know the JSON file location before we create scaladoc webpage, because there is no way to inject this URL later

I think this sentence is somehow redundant with previous `This enforce us to provide the setting while generating docs for each of the versions, however it gives us more flexibility later...` but reminds us that there is only one file globally available.

Reply:
I can try to rephrase it, but for me it seems obvious. Maybe you  see some better explanation or we should remove it to not confuse people at all.